pkg/server/plugin/zmq/broker.go

```diff
@@ -48,23 +50,37 @@ const (
 	// It is an addition to original PPP to shorten the time needed for
 	// broker to detect a normal shutdown of worker.
 	Shutdown = "\003"
+	Request = "REQ"
+	Response = "RES"
 )
 
-// parcel is used to multiplex the chan with zmq worker
-type parcel struct {
-	respChan chan []byte
-	frame    []byte
+// Parcel is used to multiplex the chan with zmq worker
+type Parcel struct {
+	Worker   string
+	RespChan chan []byte
+	Frame    []byte
 	retry    int
 }
 
-func newParcel(frame []byte) *parcel {
-	return &parcel{
-		respChan: make(chan []byte),
-		frame:    frame,
+func newParcel(frame []byte) *Parcel {
+	return &Parcel{
+		Worker:   "",
+		RespChan: make(chan []byte),
+		Frame:    frame,
 		retry:    0,
 	}
 }
 
+var requestIDPrefix = ""
```

Comment:
The python worker ID are designed to identify the worker. Here the request ID are design to identify request.

In python, the prefix is global to the process. The suffix is global to the thread. So in skygear-server, we can just look at the worker related log and know which process x thread that log is belonging to.

So in here, the request ID should do the same. By looking at the log, we will know which nested request cycle a particular request is belong to. The prefix generate on skygear-server accepting request from SDK.